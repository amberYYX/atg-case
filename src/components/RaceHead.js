import React from 'react'

const RaceHead = ({ info, id }) => {
  const changeDateFormat = stringDate => {
    const date = new Date(stringDate)
    return date.toDateString()
  }

  return (
    <div className="bg-blue-500 px-4 py-3 sm:grid sm:grid-cols-10 sm:gap-4 sm:px-6">
      <dt className="sm:col-span-2 text-5xl font-medium text-orange-100">
        {id}-{info.number}
      </dt>
      <dd className="mt-10 text-sm leading-5 text-gray-900 sm:mt-0 sm:col-span-6">
        <ul>
          <li className="pl-3 pr-4 py-3 flex items-center justify-between text-sm leading-5">
            <div className="w-0 flex-1 flex items-center text-base font-mono font-bold text-gray-100">
              {info.name ? info.name : info.status.toUpperCase()}
            </div>
          </li>
          <li className="border-t border-gray-200 pl-3 pr-4 py-3 flex items-center justify-between text-sm leading-5">
            <div className="w-0 flex-1 flex text-gray-100 items-center">
              <strong>{changeDateFormat(info.startTime)}</strong> &#12288;
              {info.startTime.split('T')[1]}
            </div>
          </li>
        </ul>
      </dd>
    </div>
  )
}

export default RaceHead
