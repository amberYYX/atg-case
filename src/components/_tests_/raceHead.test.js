import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import RaceHead from '../RaceHead'

describe('RaceHead', () => {
  const testInfo = {
    number: 6,
    name: 'game name',
    status: 'upcoming',
    startTime: '2020-11-13T19:55:00'
  }

  const id = 'V4'

  it('renders default correctly', () => {
    const tree = renderer.create(<RaceHead info={testInfo} id={id} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders content correctly with game name', () => {
    const component = render(<RaceHead info={testInfo} id={id} />)
    expect(component.container).toHaveTextContent(6)
    expect(component.container).toHaveTextContent('V4')
    expect(component.container).toHaveTextContent('game name')
    expect(component.container).toHaveTextContent('Fri Nov 13 2020 19:55:00')
  })

  const testInfo_without_game_name = {
    number: 6,
    status: 'upcoming',
    startTime: '2020-11-13T19:55:00'
  }

  it('renders content correctly with no game name', () => {
    const component = render(
      <RaceHead info={testInfo_without_game_name} id={id} />
    )
    expect(component.container).toHaveTextContent(6)
    expect(component.container).toHaveTextContent('V4')
    expect(component.container).toHaveTextContent('UPCOMING')
    expect(component.container).toHaveTextContent('Fri Nov 13 2020 19:55:00')
  })
})
