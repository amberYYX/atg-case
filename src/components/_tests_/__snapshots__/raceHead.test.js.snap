// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaceHead renders default correctly 1`] = `
<div
  className="bg-blue-500 px-4 py-2 sm:grid sm:grid-cols-10 sm:gap-4 sm:px-6"
>
  <dt
    className="sm:col-span-2 text-5xl font-medium text-orange-100"
  >
    V4
    -
    6
  </dt>
  <dd
    className="pt-1 text-sm leading-5 text-gray-900 sm:mt-0 sm:col-span-6"
  >
    <ul>
      <li
        className="pl-3 pr-4 py-3 flex items-center justify-between text-sm leading-5"
      >
        <div
          className="w-0 flex-1 flex items-center text-base font-bold -mb-2 text-blue-800"
        >
          game name
        </div>
      </li>
      <li
        className="border-t border-gray-200 pl-3 pr-4 py-3 flex items-center justify-between text-sm leading-5"
      >
        <div
          className="w-0 flex-1 flex -mt-2 text-blue-900 items-center"
        >
          <strong>
            Fri Nov 13 2020
          </strong>
           ã€€
          19:55:00
        </div>
      </li>
    </ul>
  </dd>
</div>
`;
